source/framework/core/src/TRestDataSet.cxx

```diff
@@ -894,12 +894,43 @@ TRestDataSet& TRestDataSet::operator=(TRestDataSet& dS) {
     fFilterEqualsTo = dS.GetFilterEqualsTo();
     fQuantity = dS.GetQuantity();
     fTotalDuration = dS.GetTotalTimeInSeconds();
-    fFileSelection = dS.GetFileSelection();
     fCut = dS.GetCut();
 
     return *this;
 }
 
+///////////////////////////////////////////////
+/// \brief This function merge different TRestDataSet
+/// metadata in current dataSet
+///
+Bool_t TRestDataSet::Merge(TRestDataSet& dS) {
```

Comment:
```suggestion
Bool_t TRestDataSet::Merge(const TRestDataSet& dS) {
```

We don't want a mutable reference here right? The input should be read-only (if I understood correctly).

Reply:
Indeed, should be const

Done